docs/backend-system/architecture/05-extension-points.md

```diff
@@ -25,6 +29,54 @@ export const scaffolderActionsExtensionPoint =
   });
 ```
 
-#### Registering an Extension Point
+## Registering an Extension Point
+
+For modules to be able to use your extension point, and implementation of it must be registered by the plugin. This is done using the `registerExtensionPoint` method in the `register` callback of the plugin definition.
+
+```ts
+class ActionsExtension implements ScaffolderActionsExtensionPoint {
+  addActions(...actions: TemplateAction<any>[]): void { ... }
+
+  getRegisteredActions() { ... }
+}
+
+export const scaffolderPlugin = createBackendPlugin(
+  {
+    id: 'scaffolder',
+    register(env) {
+      const actionsExtensions = new ActionsExtension();
+      env.registerExtensionPoint(
+        scaffolderActionsExtensionPoint,
+        actionsExtensions,
+      );
+
+      env.registerInit({
+        deps: { ... },
+        async init({ ... }) {
+          const actions = actionsExtension.getRegisteredActions();
+
+          // Use the registered actions when setting up the scaffolder ...
+        },
+      });
+    },
+  },
+);
+```
+
+There are a couple of things to note here. The first is that our `ActionsExtension` class both implements the `ScaffolderActionsExtensionPoint` interface, but also has additional public methods. These methods won't be available to the modules that use this extension, but we can use them here in the implementation of our plugin.
```

Comment:
```suggestion
There are a couple of things to note here. The first is that our `ActionsExtension` class both implements the `ScaffolderActionsExtensionPoint` interface, but also has additional public methods. These methods won't be available to the modules that use this extension, but we can use them here in the implementation of our plugin. Note also that the `ActionsExtension` class is _not_ exported to the outside, since it's only for the internal use of this plugin during its setup phase.
```